packages/notebook-extension/src/index.ts

```diff
@@ -347,6 +350,152 @@ export const commandEditItem: JupyterFrontEndPlugin<void> = {
   }
 };
 
+/**
+ * A plugin that provides a execution indicator item to the status bar or toolbar.
+ */
+export const executionIndicator: JupyterFrontEndPlugin<void> = {
+  id: '@jupyterlab/notebook-extension:execution-indicator',
+  autoStart: true,
+  requires: [INotebookTracker, ILabShell, ITranslator],
+  optional: [IStatusBar, ISettingRegistry],
+  activate: (
+    app: JupyterFrontEnd,
+    notebookTracker: INotebookTracker,
+    labShell: ILabShell,
+    translator: ITranslator,
+    statusBar: IStatusBar | null,
+    settingRegistry: ISettingRegistry | null
+  ) => {
+    // Set default position to toolbar.
+    let showOnToolBar = true;
+    let showProgress = true;
+    const toolbarItemName = 'executionProgress';
+
+    if (settingRegistry) {
+      const loadSettings = settingRegistry.load(trackerPlugin.id);
+      const statusbarItem = new ExecutionIndicator(translator);
+      const labShellCurrentChanged = (
+        _: ILabShell,
+        change: ILabShell.IChangedArgs
+      ) => {
+        const { newValue } = change;
+        if (newValue && notebookTracker.has(newValue)) {
+          const panel = newValue as NotebookPanel;
+          statusbarItem.model!.attachNotebook({
+            content: panel.content,
+            context: panel.sessionContext
+          });
+        }
+      };
+
+      let statusBarDisposable: IDisposable;
+      let widgetAddedSlot: (
+        tracker: INotebookTracker,
+        panel: NotebookPanel
+      ) => void;
+
+      const updateSettings = (settings: ISettingRegistry.ISettings): void => {
+        //Remove old indicator widget on status bar
+        if (statusBarDisposable) {
+          labShell.currentChanged.disconnect(labShellCurrentChanged);
+          statusBarDisposable.dispose();
+        }
+
+        // Remove old indicator widgets on toolbar
+        notebookTracker.forEach(panel => {
```

Comment:
The integration should be changed to use the new `toolbarRegistry`.

You need to generate a custom widget for this. That widget will be the indicator probably wrapped in `UseSignal` to react to settings changes (if a setting registry exist).

Reply:
Updated in latest commits